python-package/xgboost/training.py

```diff
@@ -64,28 +63,12 @@ def _train_internal(params, dtrain,
     params = _configure_metrics(params.copy())
 
     bst = Booster(params, [dtrain] + [d[0] for d in evals])
-    nboost = 0
-    num_parallel_tree = 1
 
     if xgb_model is not None:
         bst = Booster(params, [dtrain] + [d[0] for d in evals],
                       model_file=xgb_model)
-        nboost = len(bst.get_dump())
-
-    _params = dict(params) if isinstance(params, list) else params
 
-    if 'num_parallel_tree' in _params and _params[
-            'num_parallel_tree'] is not None:
-        num_parallel_tree = _params['num_parallel_tree']
-        nboost //= num_parallel_tree
-    if 'num_class' in _params and _params['num_class'] is not None:
-        nboost //= _params['num_class']
-
-    # Distributed code: Load the checkpoint from rabit.
-    version = bst.load_rabit_checkpoint()
```

Comment:
Are `load_rabit_checkpoint()` and `save_rabit_checkpoint()` deprecated? If so, we should remove `XGBoosterLoadRabitCheckpoint()` and `XGBoosterSaveRabitCheckpoint()`.

Reply:
We should, except for some tests on jvm package rely on it.  When deprecating the single point model recovery, I tried removing the rabit version counter but I failed with some tests on jvm side.  Need to spend some time on eradicating it.